examples/external-data/src/mock-external-data-service/service.ts

```diff
@@ -146,6 +146,63 @@ export async function initializeExternalDataService(props: ServiceProps): Promis
 		}
 	});
 
+	interface UnregisterWebhookRequest extends Request {
+		body: {
+			// The target URL to unsubscribe from change notifications
```

Comment:
Nit
```suggestion
			/**
			 * The target URL to unsubscribe from change notifications.
			 */
```

Reply:
Also, since we expect the URL to contain information about the task list, we should document those expectations here,

Reply:
It seems like in one place, we expected the actual request url being hit to have the task list as a url parameter.
Tests seem to expect it within the subscriber url

Where do we want this to live? I actually updated the code in this PR to be consistent in wanting it in the task list as a param in the url itself. I think it makes sense to not mix the subscriber url and the task list together but I might be missing something here

Reply:
I think the documentation on `register-for-webhook` is outdated. If you look at the code, it's reading it from the URL. Or are you seeing a place where the code is actually expecting it as a separate parameter?

Reply:
1. [`/register-for-webhook` expects it to be within the subscriber url (the url within the post body) ](https://github.com/microsoft/FluidFramework/blob/main/examples/external-data/src/mock-external-data-service/service.ts#L129-L131)
2. `[`/set-tasks/` expects it within the express request object's `request.params` (not a query search param, but a query param)](https://github.com/microsoft/FluidFramework/blob/main/examples/external-data/src/mock-external-data-service/service.ts#L210C2-L210C65)`
3.[ Similarly, `/fetch-tasks` expects it in the url  as a express request param](https://github.com/microsoft/FluidFramework/blob/main/examples/external-data/src/mock-external-data-service/service.ts#L166)
4.  [Finally, tests show that that `register-for-webhook` expects the parameter as a query search param](https://github.com/microsoft/FluidFramework/blob/main/examples/external-data/src/mock-external-data-service/service.ts#L166)

Reply:
I think the link for `register-for-webhook` tests was supposed to be [this one](https://github.com/microsoft/FluidFramework/blob/main/examples/external-data/tests/externalDataService.test.ts#L130-L142), but you're correct. The tests seems to be wrong in the sense that they add a query string param that's not actually used, we can clean that up.

I think for `fetch-tasks` and `set-tasks` we originally thought of them in a more RESTful way (routes along the lines of `/<kind of object>/<id>`), even if they're not strictly RESTful. `(un)register-for-webhook` is even less RESTful IMO (which I don't have an issue with), and in that case we need several pieces of data; the subscriber URL, and the task list id. I agree that it'd probably be best if we followed the same approach for all routes (which IMO should be to put all bits of data as separate fields in the body of the request, particularly so we didn't have to extract the task list id from the URL) but (a) I'd do it in a separate PR, and (b) if we did it at all; for a sample app I don't mind it that much. Also there's the fact that `/fetch-tasks` being a GET request cannot have a body, so that one is forced to have its parameters in the URL path or the query string.

I'd say clean up the unnecessary querystring param from the tests for clarity, and for `unregister-webhook` take the URL and the tasklist id in the body. However, note that the URL also needs to have the task list, because the caller should specify _exactly_ where they want the webhook request. We shouldn't try to build the final URL ourselves by combining the URL from the request and the task list id also from the request. Effectively, they have to give us a URL that we treat as a black box, and a task list id (which we need for our internal mappings). If on their end the URL also needs to have the task list id, they have to put it there too.